Libraries/CO.CDP.WebApi.Foundation/ResponseMiddleware.cs

```diff
@@ -0,0 +1,74 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace CO.CDP.WebApi.Foundation;
+
+internal class ResponseMiddleware(
+        RequestDelegate next,
+        ILogger<ResponseMiddleware> logger,
+        Dictionary<Type, (int, string)> exceptionMap)
+{
+    public async Task Invoke(HttpContext context)
+    {
+        var requestUrl = context.Request.GetDisplayUrl();
+
+        try
+        {
+            await next.Invoke(context);
+
+            var statusCode = context.Response.StatusCode;
+
+            if (statusCode >= 400 && statusCode < 500)
```

Comment:
When is this triggered? Is it when an error response is returned from a controller explicitly?

Reply:
yes, like bad request, unprocessable entity, unauthorized, forbidden